src/CommandingOfficers/AWBW/NRA/Greyfield.java

```diff
@@ -0,0 +1,207 @@
+          + "+10/40 stats for naval units, copters, and multirole attack planes (Stealths/Seaplanes)\n"));
+      addSharedPages();
+    }
+    protected void addSharedPages()
+    {
+      infoPages.add(new InfoPage(new HighCommand(null, null),
+            "No special effect.\n"
+          + "+10 attack and defense.\n"));
+      infoPages.add(new InfoPage(new SupplyChain(null, null),
+            "Resupply ammo, fuel, and materials.\n"
+          + "+10 attack and defense.\n"));
+      infoPages.add(AWBW_MECHANICS_BLURB);
+    }
+    @Override
+    public Commander create(GameScenario.GameRules rules)
+    {
+      return new Greyfield(this, rules);
+    }
+  }
+
+  static final long BOOST_MASK_ANY = UnitModel.SEA | UnitModel.HOVER; // Also some other unit, via below
+  final ArrayList<UnitModel> buffablePlanes = new ArrayList<>();
+  public Greyfield(CommanderInfo info, GameScenario.GameRules rules)
+  {
+    super(info, rules);
+
+    calcBuffables();
+
+    CommanderAbility.CostBasis cb = getGameBasis();
+    addCommanderAbility(new HighCommand(this, cb));
+    addCommanderAbility(new SupplyChain(this, cb));
+  }
+  protected void calcBuffables()
+  {
+    boolean matchOnAny = false;
+    long multiRoleJets = UnitModel.JET | UnitModel.AIR_TO_AIR | UnitModel.AIR_TO_SURFACE | UnitModel.ASSAULT;
+    buffablePlanes.addAll(getAllModels(multiRoleJets, matchOnAny));
+  }
+
+  @Override
+  public void modifyUnitAttack(StrikeParams params)
+  {
+    if( params.attacker.model.isAny(BOOST_MASK_ANY)
+        || buffablePlanes.contains(params.attacker.model) )
+      params.attackPower += 10;
+  }
+  @Override
+  public void modifyUnitDefenseAgainstUnit(BattleParams params)
+  {
+    if( params.defender.model.isAny(BOOST_MASK_ANY)
+        || buffablePlanes.contains(params.defender.model) )
+      params.defenseSubtraction += 40;
+  }
+
+  private static class HighCommand extends AWBWAbility
+  {
+    private static final long serialVersionUID = 1L;
+    private static final String NAME = "High Command";
+    private static final int COST = 1;
+    
+    HighCommand(Commander commander, CostBasis basis)
+    {
+      super(commander, NAME, COST, basis);
+    }
+  }
+
+  private static class SupplyChain extends AWBWAbility
+  {
+    private static final long serialVersionUID = 1L;
+    private static final String NAME = "Supply Chain";
+    private static final int COST = 3;
+
+    SupplyChain(Commander commander, CostBasis basis)
+    {
+      super(commander, NAME, COST, basis);
+    }
+
+    @Override
+    protected void perform(MapMaster gameMap)
+    {
+      super.perform(gameMap);
+      for( Unit unit : myCommander.army.getUnits() )
+      {
+        unit.resupply();
+        unit.materials = unit.model.maxMaterials;
+      }
+    }
+  }
+
+  /**
+   * Exactly the same as the other Greyfield, except he builds and buffs only Seaplanes D2D instead of any multirole jet.
+   */
+  public static class SeaplaneAbsolutist extends Greyfield
```

Comment:
But why?

Reply:
It would be friendly to users to have some justification in the CO description to explain the motivation for having two versions of Greyfield (and Sonja BW. You also mentioned the terrain stars difference for Sonja AWDSBW).